import numpy as np
from canon.exception import CanonException